### PR TITLE
feat(Examples): Add options for usage of solenoid field

### PR DESCRIPTION
**Pull request by XiaocongAi**
This PR add the magnetic field options for usage of solenoid field for the examples.

### CODE REVIEW

**Comment by XiaocongAi** [comment]
Hi @paulgessinger, could you have a look?

**Comment by XiaocongAi** [comment]
Hi @paulgessinger , what values for the `bf-solenoid-map-nbins` do you suggest to use?

**Comment by paulgessinger** [comment]
i didn't closely study the bin effect, but the benchmark i wrote uses this:
```cpp
const size_t nBinsR = 150;
const size_t nBinsZ = 200;
```

**Comment by XiaocongAi** [comment]
> i didn't closely study the bin effect, but the benchmark i wrote uses this:
> 
> ```c++
> const size_t nBinsR = 150;
> const size_t nBinsZ = 200;
> ```
Thanks! I made the update for the parameters.

File: Examples/Detectors/MagneticField/src/MagneticFieldOptions.cpp [hunk 1]
```diff
@@ -175,6 +198,39 @@ ActsExamples::Options::readMagneticField(const Variables& vars) {
...
+    std::pair<double, double> rlim, zlim;
+    getRange("bf-solenoid-map-rlim", Acts::UnitConstants::mm, rlim.first,
+             rlim.second);
+    getRange("bf-solenoid-map-zlim", Acts::UnitConstants::mm, zlim.first,
+             zlim.second);
+    const auto nbins = vars["bf-solenoid-map-nbins"].as<Reals<2>>();
+    auto mapper = Acts::solenoidFieldMapper(rlim, zlim, {nbins[0], nbins[1]},
+                                            solenoidField);
+    InterpolatedMagneticField2::Config cfg(std::move(mapper));
+    return std::make_shared<InterpolatedMagneticField2>(std::move(cfg));
```
Comment (robertlangenberg) [inline Examples/Detectors/MagneticField/src/MagneticFieldOptions.cpp:231]: in principle I approve of zealous programming, such as moving everything that isn't used. I also see it's absolutely consistent with the rest of the code (so just leave it as is), I just want to mention that variables that aren't mentioned again are optimized out for optimization >= O1, so moving is not actually doing anything here.

File: Examples/Detectors/MagneticField/src/MagneticFieldOptions.cpp [hunk 2]
```diff
@@ -66,6 +68,27 @@ void ActsExamples::Options::addMagneticFieldOptions(Description& desc) {
       "option only needs to be set if the field value unit in the field map "
       "file is not `Tesla`. The value must scale from the stored unit to the "
       "equvalent value in `Tesla`.");
+  opt("bf-solenoid-mag-tesla", value<double>()->default_value(0.),
```
Comment (robertlangenberg) [inline Examples/Detectors/MagneticField/src/MagneticFieldOptions.cpp:71]: is 0T the desired default?

Comment (XiaocongAi) [reply Examples/Detectors/MagneticField/src/MagneticFieldOptions.cpp:71]: Not desirable, but in default, the magnetic field is zero. And we only created the solenoid field when this value is not zero.

File: Examples/Detectors/MagneticField/src/MagneticFieldOptions.cpp (hunk 1 above)
Comment (XiaocongAi) [reply Examples/Detectors/MagneticField/src/MagneticFieldOptions.cpp:231]: OK. Thank you for pointign that out! We should keep this in mind when writing new code.

Comment (paulgessinger) [reply Examples/Detectors/MagneticField/src/MagneticFieldOptions.cpp:231]: If it doesn't produce a warning, it most likely doesn't hurt doing the extra move.

Comment (robertlangenberg) [reply Examples/Detectors/MagneticField/src/MagneticFieldOptions.cpp:231]: It certainly doesn't hurt, it's just making it a bit more complicated than necessary. I'm guilty of doing the same in the past. That's why I also didn't request it changed.

**Review by robertlangenberg — APPROVED** [review]
Looks all good to me!

**Pull request merged**